### PR TITLE
6811 Fix targets with property "passesIfGroupCount.gte" counting 'pass:false'

### PR DESCRIPTION
**Pull request by njogz**
# Description

Fix a bug where the incorrect target was being shown for targets that use `passesIfGroupCount.gte`

medic/cht-core#6811

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: shared-libs/rules-engine/test/target-state.spec.js [hunk 1]
```diff
@@ -181,7 +181,8 @@ describe('target-state', () => {
       targetState.storeTargetEmissions(state, [], [
         mockEmission({ _id: 'a', groupBy: '1', pass: true }),
         mockEmission({ _id: 'b', groupBy: '2', pass: false }), // pass should have no effect
-        mockEmission({ _id: 'c', groupBy: '1' }),
+        mockEmission({ _id: 'c', groupBy: '1', pass: true }),
+        mockEmission({ _id: 'd', groupBy: '1', pass: false }),
```
Comment (dianabarsan) [inline shared-libs/rules-engine/test/target-state.spec.js:185]: Please add another group here that has 2 emissions, one passing and one failing, to prove that this group doesn't count as passed.

**Review by dianabarsan — CHANGES_REQUESTED** [review]
Nice work! Thanks for fixing this. I left a couple of suggestions inline.

File: shared-libs/rules-engine/src/target-state.js [hunk 2]
```diff
@@ -125,20 +125,26 @@ module.exports = {
...
         .filter(emission => emission.groupBy)
-        .reduce((agg, curr) => {
-          if (!agg[curr.groupBy]) {
-            agg[curr.groupBy] = 0;
-          }
-
-          agg[curr.groupBy]++;
-          return agg;
-        }, {});
+        .reduce(reducer, {});
```
[the remark this answers is not in the document]
Comment (njogz) [reply shared-libs/rules-engine/src/target-state.js]: Nice! I have made the change.

**Comment by njogz** [comment]
@dianabarsan thanks for the suggestions. I have made the requested changes. Please have another look when you can.

**Review by dianabarsan — APPROVED** [review]
Great work!

**Pull request merged**